### PR TITLE
refactor:(std/datetime) weekOfYear: Use constants, improve readability

### PR DESCRIPTION
**Pull request by jsejcksn**
Related issue: #6601
Related PR: #6659

This PR improves readability for code and documentation.

I noticed the use of literal millisecond and integer values, and an opportunity to remove some comments and improve code readability, so I refactored it, using constants instead of the literals.

The new tests are sourced from moment.js.

### CODE REVIEW

**Comment by jsejcksn** [comment]
While working on this, I felt a pull to add a `currentWeekOfYear` function to mirror the `currentDayOfYear` function. However, I'm not sure what real value that would add, since it's a very simple abstraction—literally:

```ts
weekOfYear(new Date());
```

This made me wonder further: what is the defined purpose of `std/datetime`? These kinds of functions already exist elsewhere in dedicated libraries like `date-fns`, `moment`, `luxon`, etc. However, the style guide states [_Do not depend on external code_](https://github.com/denoland/deno/blob/master/docs/contributing/style_guide.md#do-not-depend-on-external-code), so is the aim to re-invent the wheel or copy from them?

File: std/datetime/mod.ts [hunk 1]
```diff
@@ -8,6 +8,17 @@ export const MINUTE = SECOND * 60;
...
+
+enum Day {
+  Sun,
+  Mon,
+  Tue,
+  Wed,
+  Thu,
+  Fri,
+  Sat,
+}
```
Comment (jsejcksn) [inline std/datetime/mod.ts:21]: Should these be exported, or kept private unless/until needed? I didn't see any guidelines about conditions for exporting, keeping exports to a minimal, etc.

**Comment by jsejcksn** [comment]
Re: fmt

I ran:

```shell
./tools/format.py
```

from the repo root directory, but it exited with an error. I don't remember the exact error, but I'm wondering if that step can be added to the repo actions, since [I'm not the first to run into this problem](https://github.com/denoland/deno/pull/6659#issuecomment-656252869).

Related: Is `deno fmt` equivalent to the python formatter for TS/JS files?

**Comment by bartlomieju** [comment]
> This made me wonder further: what is the defined purpose of `std/datetime`? These kinds of functions already exist elsewhere in dedicated libraries like `date-fns`, `moment`, `luxon`, etc. However, the style guide states [_Do not depend on external code_](https://github.com/denoland/deno/blob/master/docs/contributing/style_guide.md#do-not-depend-on-external-code), so is the aim to re-invent the wheel or copy from them?

The purpose for `datetime/` hasn't really been defined. It's definitely useful to have date manipulation utilities in standard library, but the scope hasn't been set. Before stabilizing `std/` we'll reevaluate each module. 

> from the repo root directory, but it exited with an error. I don't remember the exact error, but I'm wondering if that step can be added to the repo actions, since I'm not the first to run into this problem.

This issue should be resolved when #6682 lands.

> Related: Is deno fmt equivalent to the python formatter for TS/JS files?

No, it's not, current repository uses `prettier`.

File: std/datetime/mod.ts (hunk 1 above)
Comment (timreichen) [reply std/datetime/mod.ts:21]: I suggested to remove the ```currentDayOfYear``` in [that issue](https://github.com/denoland/deno/issues/6601).

As for the purpose of ```std/datetime```: I think a clear definition has to be worked out still, but I think it is a good addition as a std library. In my view the module has theses purposes at the moment:
- provide commonly used functions related to date and time
- provide datetime parser and formatter

I think some 3rd party libraries do the one, some the other.
I did a PR to add a generic parser and a formatter.  In a sense some features are reinventing the wheel (like parser and formatter) but the attention lies to provide compatibility with ```Intl```.
Also this module will likely change once this [proposal](https://github.com/tc39/proposal-temporal) is at stage 4.

Comment (jsejcksn) [reply std/datetime/mod.ts:21]: `Temporal` looks like a step forward. 🧗 

@bartlomieju Any feedback about exports?

**Pull request merged**